src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -8052,7 +8052,6 @@ and TcNameOfExpr cenv env tpenv (synArg: SynExpr) =
                 | Result (_, item, _, _, _ as res)
                     when
                          (match item with
-                          | Item.Types _
```

Comment:
Above, it says
> // However we don't commit for a type names - nameof allows 'naked' type names and thus all type name
>// resolutions are checked separately in the next step.

So, probably should be handled in `resolvedToTypeName ` instead?